### PR TITLE
add semantic conventions for specification v1.16.0

### LINKED ISSUE #6714: add semantic conventions for specification v1.16.0
**Is your feature request related to a problem? Please describe.**
I would like to use sem-convs from spec v1.16.0.

**Describe the solution you'd like**
add semantic conventions for specification v1.16.0

### PR DESCRIPTION
**Pull request by frzifus**
**Description:** 
Add semantic conventions for specification [v1.16.0](https://github.com/open-telemetry/opentelemetry-specification/releases/tag/v1.16.0).

**Link to tracking Issue:** 
Belongs to https://github.com/open-telemetry/opentelemetry-collector/issues/6696#issuecomment-1343135506
Closes #6714

**Additional:** 

Convention type `trace` trace leads to the following output:
<details>

```
panic: unable to render template: exit status 1
Traceback (most recent call last):
  File "/usr/local/bin/gen-semconv", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/opentelemetry/semconv/main.py", line 67, in main
    semconv = parse_semconv(args, parser)
  File "/usr/local/lib/python3.8/site-packages/opentelemetry/semconv/main.py", line 38, in parse_semconv
    semconv.finish()
  File "/usr/local/lib/python3.8/site-packages/opentelemetry/semconv/model/semantic_convention.py", line 317, in finish
    fixpoint_ref = self.resolve_ref(semconv)
  File "/usr/local/lib/python3.8/site-packages/opentelemetry/semconv/model/semantic_convention.py", line 458, in resolve_ref
    raise ValidationError.from_yaml_pos(
opentelemetry.semconv.model.exceptions.ValidationError: Semantic Convention trace-exception reference `exception.type` but it cannot be found! - @2:5
```

</details>

~According to the [sem-conv changelog](https://github.com/open-telemetry/build-tools/blob/v0.14.0/semantic-conventions/CHANGELOG.md), there should have been no breaking change. The last one was in version [v0.12.0](https://github.com/open-telemetry/build-tools/blob/v0.14.0/semantic-conventions/CHANGELOG.md#v0120) and was resolved [here](https://github.com/open-telemetry/opentelemetry-collector/pull/6213/files#diff-e8e0fbe7ca5796b64b1c0293e112c6c2eb012bf174a4856ccc78bd04c3f9bf73R19-R26).~

~Can someone help/show me what to look for?~

---

**update**

Thanks for the hint from @Aneurysm9. He noticed that exception was moved from tracing[[1](https://github.com/open-telemetry/opentelemetry-specification/pull/2819/files#diff-663bb187428ca23e3c20d7ee34a08d14a4f0a718db80fddb5b3c2e1244d49203)][[2](https://github.com/open-telemetry/opentelemetry-specification/pull/2819/files#diff-f69ca056549052e4fab19634918c00ca992d54b0b48c8e159095b13dd8f68d94R1)]. Now I could now generate trace and resources as follows:

```bash
$ docker run --rm -v ${PWD}/tmp:/data:z otel/semconvgen:0.14.0 --yaml-root /data/input/semantic_conventions --only resource code --template /data/template.j2 --output /data/output/generated_resource.go --parameters conventionType=resource
$ docker run --rm -v ${PWD}/tmp:/data:z otel/semconvgen:0.14.0 --yaml-root /data/input/semantic_conventions --only span code --template /data/template.j2 --output /data/output/generated_trace.go --parameters conventionType=trace
```

### CODE REVIEW

**Comment by frzifus** [comment]
any updates here @open-telemetry/collector-approvers ? :smiley:

**Review by dmitryax — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Comment by frzifus** [comment]
Is something else needed? :)

**Review by Aneurysm9 — CHANGES_REQUESTED** [review]
It looks like you had to change the invocation used to generate these constants, but have not updated the `gensemconv` make target.  That needs to be done as well.  If changes are required in the generator then they should be made first and the dependency updated.

```console
$ make gensemconv SPECPATH=../opentelemetry-specification SPECTAG=v1.16.0
Generating semantic convention constants from specification version v1.16.0 at ../opentelemetry-specification
semconvgen -o semconv/v1.16.0 -t semconv/template.j2 -s v1.16.0 -i ../opentelemetry-specification/semantic_conventions/resource -p conventionType=resource -f generated_resource.go
semconvgen -o semconv/v1.16.0 -t semconv/template.j2 -s v1.16.0 -i ../opentelemetry-specification/semantic_conventions/trace -p conventionType=trace -f generated_trace.go
panic: unable to render template: exit status 1
Traceback (most recent call last):
  File "/usr/local/bin/gen-semconv", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/opentelemetry/semconv/main.py", line 67, in main
    semconv = parse_semconv(args, parser)
  File "/usr/local/lib/python3.8/site-packages/opentelemetry/semconv/main.py", line 38, in parse_semconv
    semconv.finish()
  File "/usr/local/lib/python3.8/site-packages/opentelemetry/semconv/model/semantic_convention.py", line 317, in finish
    fixpoint_ref = self.resolve_ref(semconv)
  File "/usr/local/lib/python3.8/site-packages/opentelemetry/semconv/model/semantic_convention.py", line 458, in resolve_ref
    raise ValidationError.from_yaml_pos(
opentelemetry.semconv.model.exceptions.ValidationError: Semantic Convention trace-exception reference `exception.type` but it cannot be found! - @2:5


goroutine 1 [running]:
main.main()
        /Users/mirabant/go/pkg/mod/go.opentelemetry.io/build-tools/semconvgen@v0.3.0/generator.go:61 +0x44e
make: *** [gensemconv] Error 2
```

**Comment by frzifus** [comment]
Thanks for the feedback @Aneurysm9. I have adjusted the `Makefile` rule. But to use the rule, this [semconvgen patch](https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/217) is needed.

File: Makefile
Comment (Aneurysm9) [inline Makefile:291]:
```suggestion
	semconvgen -o semconv/${SPECTAG} -t semconv/template.j2 -s ${SPECTAG} -i ${SPECPATH}/semantic_conventions --only=resource -p conventionType=resource -f generated_resource.go
	semconvgen -o semconv/${SPECTAG} -t semconv/template.j2 -s ${SPECTAG} -i ${SPECPATH}/semantic_conventions --only=span -p conventionType=trace -f generated_trace.go
```

**Comment by Aneurysm9** [comment]
Once the generator change is landed this will need to update `internal/tools/go.mod` to ensure the use of the newer generator.  Otherwise, LGTM.

**Comment by frzifus** [comment]
@Aneurysm9 I updated the internal semconvgen to the latest version.

**Comment by Aneurysm9** [comment]
> @Aneurysm9 I updated the internal semconvgen to the latest version.

I just made a `v0.4.0` release.  Can you update to that quickly?

**Review by Aneurysm9 — APPROVED** [review]

**Comment by frzifus** [comment]
done :+1:

**Review by dmitryax — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**
